tests/pos/i14013.scala

```diff
@@ -0,0 +1,15 @@
+object Foo {
+  case class Bar(i: Int)
+
+  private implicit class BarOps(bar: Bar) {
```

Comment:
This was marked as `scala2-compat`, is this behavior something we also want with `given`s?

Should this one be visible in `class Foo`?
```scala
private given BarOps: AnyRef with {
    extension (bar: Bar)
      def twice: Bar = Bar(bar.i * 2)
  }
```